### PR TITLE
Ignite 1481

### PR DESCRIPTION
**Pull request by kcheng-mvp**
https://issues.apache.org/jira/browse/IGNITE-1481

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java [hunk 1]
```diff
@@ -17,37 +17,9 @@
 
 package org.apache.ignite.internal.processors.cache;
 
-import java.io.Serializable;
```
Comment (dmagda) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java]: Please change settings in your IDE that affect on how classes are imported - either the whole package or a particular class.

Recently Ignite community switched from 'package' based imports to a per class based import. So you shouldn't remove all these imports.

To make things simpler please set {ignite}/idea/ignite_codeStyle.xml that will setup everything for you.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java [hunk 2]
```diff
@@ -371,7 +339,7 @@ private void validate(IgniteConfiguration c,
 
         if (cc.getCacheMode() == LOCAL && !cc.getAffinity().getClass().equals(LocalAffinityFunction.class))
             U.warn(log, "AffinityFunction configuration parameter will be ignored for local cache [cacheName=" +
-                U.maskName(cc.getName()) + ']');
+                U.maskName(cc.getName()) + ']'+", this check will be moved to initialize() method");
```
Comment (dmagda) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java]: Please remove this change from here.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java [hunk 3]
```diff
@@ -252,6 +217,9 @@ else if (cfg.getCacheMode() == REPLICATED) {
                     if (aff.getHashIdResolver() == null)
                         aff.setHashIdResolver(new AffinityNodeAddressHashResolver());
                 }
+            } else if (cfg.getCacheMode() == LOCAL && !(cfg.getAffinity() instanceof LocalAffinityFunction)){
+                cfg.setAffinity(new LocalAffinityFunction());
+                U.warn(log, "Only LoalAffinityFunction() is applicable for local cache [cacheName=" + U.maskName(cfg.getName()) + ']');
```
Comment (dmagda) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java]: Please set text message to the one that starts this way

AffinityFunction configuration parameter will be ignored for local cache [cacheName...

File: modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        stopAllGrids();
+    }
+
+    /**
+     * Test Case for Bug fix :  https://issues.apache.org/jira/browse/IGNITE-1481
+     */
+    public void testWronglySetAffinityFunctionForLocalCache(){
+        Ignite node = ignite(NODE_CNT - 1);
+        CacheConfiguration ccf = node.getOrCreateCache(CACHE1).getConfiguration(CacheConfiguration.class);
+        assertEquals("org.apache.ignite.internal.processors.cache.GridCacheProcessor$LocalAffinityFunction",ccf.getAffinity().getClass().getName());
```
Comment (dmagda) [inline modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java]: You can simply check that
 assertTrue(ccf.getAffinity().getClass() instanceof LocalAffinityFunction)

File: modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    protected void afterTestsStopped() throws Exception {
+        super.afterTestsStopped();
+        stopAllGrids();
+    }
+
+    /**
+     * Test Case for Bug fix :  https://issues.apache.org/jira/browse/IGNITE-1481
```
Comment (dmagda) [inline modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java]: Remove this description please. Refer to other tests to see how we comment test messages.

File: modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        super.beforeTestsStarted();
+        startGrids(NODE_CNT);
+    }
+
+    @Override
+    protected void afterTestsStopped() throws Exception {
+        super.afterTestsStopped();
+        stopAllGrids();
+    }
+
```
Comment (dmagda) [inline modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java]: Please create additional test for a dynamic cache that is started using

ignite.getOrCreateCache(...)

File: modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi;
+import org.apache.ignite.spi.discovery.tcp.ipfinder.TcpDiscoveryIpFinder;
+import org.apache.ignite.spi.discovery.tcp.ipfinder.vm.TcpDiscoveryVmIpFinder;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+
+public class LocalAffinityFunctionTest extends GridCommonAbstractTest{
+
+    protected static TcpDiscoveryIpFinder ipFinder = new TcpDiscoveryVmIpFinder(true);
+    private static final int NODE_CNT = 1;
```
Comment (dmagda) [inline modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java]: Please take a look at Ignite coding guideline that describes line breaking, indentation and other rules and fix whenever applicable.

https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines

File: modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java (hunk 4 above)
Comment (kcheng-mvp) [inline modules/core/src/test/java/org/apache/ignite/cache/affinity/local/LocalAffinityFunctionTest.java]: As LocalAffinityFunction is a private class, can not refer it directly.

**Pull request closed without merging**